cli/js/lib.deno.unstable.d.ts

```diff
@@ -1245,4 +1245,7 @@ declare namespace Deno {
    *  Requires `allow-env` permission.
    */
   export function hostname(): string;
+
+  /** **UNSTABLE**: The URL of the file that was originally executed from the command-line. */
+  export const mainUrl: URL;
```

Comment:
Can't we return a frozen `URL` instead?

Reply:
We shouldn't get into freezing things:
1. This really does count as monkey-patching a built-in, since it overwrites specced (?) property descriptors.
2. It limits how the class has to be implemented for freezing to work.
    - ...particularly, it won't work for `URL`s because the actual state is stored in a separate hidden object, while the public fields just use getters and setters.

I think the `URL` class exists to make manipulating them easier, so I'm not sure a frozen one even makes sense.

Also, I initially suggested this after not immediately finding any web API that has a `URL` field set in this way.

Reply:
Fair points. I've changed it to a string named `mainModule`. 👍